src/main/java/com/example/demo/dto/RecipeDto.java

```diff
@@ -0,0 +1,41 @@
+package com.example.demo.dto;
+
+import com.example.demo.entity.IngredientsRecipes;
+import com.example.demo.entity.User;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Builder
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class RecipeDto {
+    private UserDto createdByDto;
+
+    @JsonProperty
+    private int creatorUserId;
+
+    @JsonProperty
+    private int recipeId;
+
+    @JsonProperty
+    private int foodId;
+
+    @JsonProperty
+    private String recipeName;
+
+    @JsonProperty
+    private boolean isPublic;
+
+    private int cookingTime;
+    private String cuisine;
+
+    private List<IngredientsRecipes> ingredientsRecipesList; // There is Food, which has proteins and so on

```

Comment:
There should not be any comments like this. If something is not clear, you can use javadoc on top of class declaration to describe fields.
https://www.dummies.com/article/technology/programming-web-design/java/how-to-use-javadoc-to-document-your-classes-153265/